src/App.jsx

```diff
@@ -0,0 +1,135 @@
+                boxShadow: "0"
+            }
+        }).showToast();
+        console.log("Empty")
+        return;
+    }
+  
+  // Save current graphic and start spinner
+  let oldWeatherGraphic = weather.graphic;
+  setWeather({ ...weather, graphic: "animated/spinner.svg"});
+  
+    fetch(`https://api.openweathermap.org/data/2.5/weather?q=${search}&appid=${API_KEY}&units=imperial`)
+    .then((response) => {
+      if (!response.ok) {
+        throw Error("Could not resolve");
+      }
+      return response.json()
+    })
+    .then(
+      (data) => {
+        console.log(data)
+        
+        setWeather({
+          graphic: getWeatherIcon(data.weather[0].main),
+          graphicAlt: `${data.weather[0].main} Icon`,
+          temperature: Math.round(data.main.temp),
+          feelsLikeTemp: Math.round(data.main.feels_like),
+          humidity: data.main.humidity,
+          wind: data.wind.speed,
+        location: `${data.name}, ${data.sys.country}`,
+        weatherDescription: uppercaseWords(data.weather[0].description),
+        })
+        
+      },
+      (error) => {
+        Toastify({
+          text: "API call failed",
+          position: "center",
+          duration: 1000,
+          style: {
+            background: "#f03551",
+            boxShadow: "0"
+        }
+      }).showToast();
+        setWeather({ ...weather, graphic: oldWeatherGraphic});
+      }
+    ).catch(err => {
+      console.log(err)
+    }
+      
+    )
+  }  
+
+  return (
+    <div className="center">
+      <div className="container">
+        <div className="search">
+          <form id="search-form" onSubmit={handleSubmit}>
+            <input 
+            id="search" 
+            type="text"
+            placeholder={search} 
+            onChange={(event) => {setSearch(event.target.value); }}/>
+            <input type="submit" />
+          </form>
+          <div id="location" className="weight-thin gray">{weather.location}</div>
+        </div>
+        <div className="weather-box">
+          <img id="weather-graphic" src={weather.graphic} alt={weather.graphicAlt}/>
+          <p id="weather-description" className="weight-thin gray">{weather.weatherDescription}</p>
+        </div>
+        <div className="temperature">
+          <h1 id="temp-value">{weather.temperature}<sup>&deg;F</sup></h1>
+          <p id="feels-like" className="weight-thin gray">Feels Like: {Math.round(weather.feelsLikeTemp)} <sup>&deg;F</sup></p>
+        </div>
+        <div className="humidity">
+          <img src={waterDroplet} alt="Water Droplet"/>
+          <p id="humidity">{weather.humidity}%</p>
+          <p className="humidity-label">Humidity</p>
+        </div>
+        <div className="wind">
+          <img src={windGust} alt="Wind Gust"/>
+          <p id="wind">{weather.wind} mp/h</p>
+          <p className="wind-label">Wind Speed</p>
+        </div>
+      </div>
+    </div>
+  )
+}
+
+export default App
```

Comment:
(🐛)

```javascript
import React, { useState, useCallback, memo } from 'react';

// ... other imports ...

const WeatherBox = memo(function WeatherBox({ weather }) {
  return (
    <div className="weather-box">
      <img
        id="weather-graphic"
        src={weather.graphic}
        alt={weather.graphicAlt}
      />
      <p id="weather-description" className="weight-thin gray">
        {weather.weatherDescription}
      </p>
    </div>
  );
});

function App() {
  // ... useState code ...

  const handleSubmit = useCallback(
    (e) => {
      // ... same as before ...
    },
    [search, weather]
  );

  // ... return JSX code ...
  return (
    <div className="center">
      <div className="container">
        {/* ... other components ... */}
        <WeatherBox weather={weather} />
        {/* ... other components ... */}
      </div>
    </div>
  );
}

export default App;
```

ChatGPT 4 said to use this so you're probably bloating this and doing it wrong.

Reply:
This doesn't include any of the actual working parts of the program, just a bunch of scaffolding with comments.